### PR TITLE
Input validation pr5

### PR DESCRIPTION
**Pull request by RainingComputers**

### CODE REVIEW

File: meiki_server/auth/auth.go [hunk 1]
```diff
@@ -65,6 +70,18 @@ func CreateAuth(ctx context.Context, token_coll *mongo.Collection, user_coll *mo
 	return Auth{token_coll, user_coll}, nil
 }
 
+func validateCredentials(username string, password string) error {
+	if !usernameRegex.MatchString(username) {
+		return ErrInvalidUsername
+	}
+
+	if len(password) < 5 {
```
Comment (alexsanjoseph) [inline meiki_server/auth/auth.go:78]: probably a separate function at some point of time, but good enough for now

File: meiki_server/auth/routes.go [hunk 2]
```diff
@@ -20,6 +19,23 @@ type SessionCredentials struct {
 	Token    string `json:"token"`
 }
 
+const MSG_INVALID_USERNAME = "Username should not contain any special characters other than '-' and '_'"
```
Comment (alexsanjoseph) [inline meiki_server/auth/routes.go:22]: Ha, nice!

**Comment by alexsanjoseph** [comment]
Great changes. TODO for later: If we get an invalid credentials response from user, we should show an alert in the UI telling how the username/password should be

File: meiki_server/auth/auth.go (hunk 1 above)
Comment (RainingComputers) [reply meiki_server/auth/auth.go:78]: Yep, I saw a lot of complex regex but would not prefer that, this is probably a seperate spike I guess

**Comment by RainingComputers** [comment]
Well it does, but it is not specific, it does not say the exact violation. So yes that is a TODO. Maybe will add this to the password spike how to handle this kinds of errors.

**Pull request merged**